core/src/main/java/org/acegisecurity/Authentication.java

```diff
@@ -49,7 +49,7 @@ public interface Authentication extends Principal, Serializable {
 
     boolean isAuthenticated();
 
-    void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException;
+    void setAuthenticated(boolean isAuthenticated);
```

Comment:
[Effective Java 3rd Edition §74](https://books.google.com/books?id=BIpDDwAAQBAJ) says:

> Use the Javadoc `@throws` tag to document each exception that a method can throw, but do not use the `throws` keyword on unchecked exceptions.

This PR removes the `throws` keyword on this unchecked exception (good), but it does not add the Javadoc `@throws` tag (bad).